packages/ui/src/components/DataStory/DataStory.tsx

```diff
@@ -8,16 +8,35 @@ import { useEffect, useRef, useState } from 'react';
 import { ReactFlowNode } from '../Node/ReactFlowNode';
 import { DataStoryCanvasProvider } from './store/store';
 import { DataStoryCanvas } from './DataStoryCanvas';
+import { Tree } from './clients/Tree';
+import { sleep } from '@data-story/core';
 
 export const DataStory = (
   props: Omit<DataStoryProps,'setSidebarKey'>
 ) => {
+  const [tree, setTree] = useState<Tree>();
   const [selectedNode, setSelectedNode] = useState<ReactFlowNode>();
   const [isSidebarClose, setIsSidebarClose] = useState(true);
   const [updateSelectedNodeData, setUpdateSelectedNodeData] = useState<ReactFlowNode['data']>();
   const [sidebarKey, setSidebarKey] = useState('');
   const partialStoreRef = useRef<Partial<StoreSchema>>(null);
 
+  useEffect(() => {
+    const { clientv2 } = props
+    if(!clientv2) return;
+    console.log('Got a clientv2')
+
+    const getTree = async () => {
+      const tree = await clientv2.workspacesApi.get({ path: '/' })
```

Comment:
1. It's generally not recommended to directly use `props` in the dependency array of `useEffect`, as any change in `props` could trigger the effect. This may include changes in props unrelated to `clientv2`, leading to unnecessary executions and impacting performance. A more precise approach would be to list `clientv2` as a dependency.
2. There's a need to implement cancellation or error handling for requests.
3. For data fetching in React, I suggest using `useRequest` from ahooks or `useSWR` from next.js.